chans/chans.go

```diff
@@ -83,3 +133,50 @@ func Push[T any](ctx context.Context, c chan<- T, x T) bool {
 		return true
 	}
 }
+
+// Range creates a channel that inclusively contains all values from `from` to `to`.
+func Range[T constraints.Integer](from, to T) <-chan T {
+	if from > to {
+		return Empty[T]()
+	}
+
+	newCh := make(chan T)
+
+	go func() {
+		defer close(newCh)
+
+		for i := from; i <= to; i++ {
+			newCh <- i
+		}
+	}()
+
+	return newCh
+}
+
+// RangeBig creates a channel that inclusively contains all values from `from` to `to`.
+func RangeBig(from, to *big.Int) <-chan *big.Int {
+	if from.Cmp(to) > 0 {
+		return Empty[*big.Int]()
+	}
+
+	newCh := make(chan *big.Int)
+
+	go func() {
+		defer close(newCh)
+
+		for i := (&big.Int{}).Set(from); i.Cmp(to) <= 0; i.Add(i, oneBig) {
+			newCh <- (&big.Int{}).Set(i)
+		}
+	}()
+
+	return newCh
+}
+
+// RangeBigExcl creates a channel that exclusively contains all values from `from` to `to`.
+// I.e. the first value will be `from+1` and the last value will be `to-1`.
+func RangeBigExcl(from, to *big.Int) <-chan *big.Int {
+	from = (&big.Int{}).Add(from, oneBig)
+	to = (&big.Int{}).Sub(to, oneBig)
+
+	return RangeBig(from, to)
+}
```

Comment:
this one seems pretty specific, I'd rather just keep it in eds

Reply:
removed - https://github.com/axelarnetwork/utils/pull/71/commits/7aee61f4f0f4a2be5fee0aba50a1a924186e7a83